core/src/main/java/com/linecorp/armeria/common/HttpResponseUtil.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletionStage;
+
+import com.linecorp.armeria.common.util.Exceptions;
+
+final class HttpResponseUtil {
+
+    static void delegateWhenStageComplete(CompletionStage<? extends HttpResponse> stage,
+                                          DeferredHttpResponse res) {
```

Comment:
Do we need to extract this into a Util class than `private static` in `HttpResponse`?

Reply:
Couldn't have a private method in interface, yet. 😢

Reply:
Ah, sorry, I realized that is java 9 feature.
Then how about using a nested static class? 😆
https://dzone.com/articles/java-8-declare-private-and-protected-methods-in-in

Reply:
As we discussed, this will expose one more class which will bring us another breaking change when we force the minimum Java version to 9, so let's just leave it as it is. 😉